### PR TITLE
Fix Kotlin file problem with CallOnRequestPermissionsResultDetector

### LINKED ISSUE #440: Generated onRequestPermissionsResult method not called
override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        // NOTE: delegate the permission handling to generated function
        onRequestPermissionsResult(requestCode, grantResults)
    }

    @NeedsPermission(Manifest.permission.CAMERA)
    override fun showCamera() {
        cameraIntent()
    }

    @NeedsPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
    override fun showGallery() {
        galleryIntent()
    }

    @OnPermissionDenied(Manifest.permission.CAMERA)
    override fun showDeniedForCamera() {
        permissionDialog("Camera permission is necessary",{ showCameraWithPermissionCheck() })
    }

    @OnPermissionDenied(Manifest.permission.READ_EXTERNAL_STORAGE)
    override fun showDeniedForGallery() {
        permissionDialog("Gallery permission is necessary",{ showGalleryWithPermissionCheck() })
    }
 
___

## Overview

"Generated onRequestPermissionsResult method not called." error.

### Expected

There is a lint error , but the file was generated and i dont know why the error appear..

### Actual

Lint error with : 
"Generated onRequestPermissionsResult method not called." error.
image

## Environment

- Which library version are you using? **3.1.0**
- On which devices do you observe the issue? **The error on android studio**
- Note any other information that might be useful

### PR DESCRIPTION
**Pull request by shiraji**
Close #440

### CODE REVIEW

**Comment by hotchemi** [comment]
Oh this is cool!

**Comment by shiraji** [comment]
Rename the PR since this only fix the problem with `CallOnRequestPermissionsResultDetector`.

Also, I will send another PR for adding more kotlin test cases for other lints.

File: lint/src/test/java/permissions/dispatcher/CallOnRequestPermissionsResultDetectorTest.java [hunk 1]
```diff
@@ -96,4 +97,45 @@ public void callOnRequestPermissionsResultDetector() throws Exception {
...
+        @Language("JAVA") String onNeeds = getOnNeedsPermission();
+
+        @Language("JAVA") String onShow = getOnRationaleAnnotation();
+
+        @Language("kotlin") String foo = ""
+                + "package permissions.dispatcher;\n"
+                + "@RuntimePermissions\n"
+                + "class Foo : android.app.Activity {\n"
+                + "fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {\n"
+                + "super.onRequestPermissionsResult(requestCode, permissions, grantResults);\n"
```
Comment (hotchemi) [inline lint/src/test/java/permissions/dispatcher/CallOnRequestPermissionsResultDetectorTest.java]: Nit: should we remove semicolon? 👀 line below is the same 🙇

Comment (shiraji) [reply lint/src/test/java/permissions/dispatcher/CallOnRequestPermissionsResultDetectorTest.java]: Nice catch! I just replaced Java to Kotlin without checking these points! Let me fix these

**Comment by shiraji** [comment]
Fixed and squashed commits

**Pull request merged**